src/algebra/order/to_interval_mod.lean

```diff
@@ -539,6 +549,66 @@ begin
   exact (mem_Ioo_mod_iff_to_Ico_mod_eq_to_Ioc_mod hb).symm,
 end
 
+/-- `to_Ico_div` disagrees with `to_Ioc_div` if `x` lies on a boundary. -/
+lemma not_mem_Ioo_mod_iff_to_Ico_div_eq_to_Ioc_div_add_one (a : α) {b : α} (hb : 0 < b) (x : α) :
+  ¬mem_Ioo_mod a b x ↔ to_Ico_div a hb x = to_Ioc_div a hb x + 1 :=
```

Comment:
This should simply be the negation of [mem_Ioo_mod_iff_to_Ico_div_add_one_ne_to_Ioc_div](https://leanprover-community.github.io/mathlib_docs/algebra/order/to_interval_mod.html#mem_Ioo_mod_iff_to_Ico_div_add_one_ne_to_Ioc_div) (unfortunately, doc-gen hasn't picked up the changes in your #18105 yet.)

Reply:
Nice catch, I've reproved it that way. I think all of the lemmas with an `ne` on one side are worth having the negated version of with an `=`; so I added a couple more.